### ISSUE TITLE
Resizing screen doesn't move windows, which can leave them largely non-visible

### ISSUE THREAD

**Issue by lurch** [labwc/labwc#1476]
I'm using Labwc (on Raspberry Pi 4 - I'm one of @spl237 's colleagues) and my HDMI monitor has a native resolution of 2560x1440.
I can launch `arandr` and it then appears in the centre of the screen (as all newly-launched applications do). If I change the resolution to 1280x1024 and click "Apply", then the screen-resolution-change happens, but the XY offset of the arandr window doesn't change, which means that most of the window is now hidden off the right of the screen, and I'm unable to see the OK / Cancel buttons on the dialog prompting me to confirm the new screen resolution. As shown in this screenshot:
![change_resolution_labwc](https://github.com/labwc/labwc/assets/476186/6823823e-effe-421e-b519-58622dbf7ace)

If I try the same thing in Wayfire, the arandr window gets moved after (during?)  the screen-resolution-change so that it's still fully-visible:
![change_resolution_wayfire](https://github.com/labwc/labwc/assets/476186/5fcccea7-3a20-497f-8e35-6f78550b1c22)

**Comment by ahesford** [comment]
Can you try this on the master branch? Last-layout restoration may help, but I have a suspicion that it won't. We should probably unconditionally call `update_last_layout_geometry(view)` in `view_adjust_for_layout_change` rather than waiting until a view loses its output.

**Comment by lurch** [comment]
The `master` branch exhibits identical behaviour.

**Comment by spl237** [comment]
For future reference, we are using the master branch anyway - at present, I am updating our package from it every couple of days!

**Comment by ahesford** [comment]
The issue here is that we consider a window that is onscreen by even one pixel as good enough to not trigger repositioning. That makes sense when you're restoring from maximized/fullscreen/tiled views (presumably, only one pixel is visible because you put it there), but doesn't really make sense when the layout has changed. Let's make the "visibility" requirement stricter in those cases to get better behavior.

#1478

**Comment by Consolatis** [comment]
Should be fixed on latest master. Thanks for bringing this up.

**Comment by spl237** [comment]
> Regarding `arandr` - I had forgotten about that tool. How do you make it work on Wayland? Have you done a `s/xrandr/wlr-randr/` patch or is it actually supported upstream?

I have a very heavily modified version which supports xrandr for openbox, and wlr-randr for both wayfire and labwc. Still work in progress - it's a bit of a hack and I am currently in the process of a major refactoring to tidy it up - see https://github.com/raspberrypi-ui/arandr

**Comment by johanmalm** [comment]
> I have a very heavily modified version which supports xrandr for openbox, and wlr-randr for both wayfire and labwc. Still work in progress - it's a bit of a hack but I am currently in the process of a major refactoring to tidy it up - see https://github.com/raspberrypi-ui/arandr/tree/refactor

Oh nice. Obviously a bit more modding required that I imagined. I suspect a lot of people will be interested in that :smile:

**Comment by spl237** [comment]
It's a lot better now than it was 48 hours ago; it's the first time I've had a chance to sit down and clean up all the quick-and-dirty hacks I've added over the last couple of years - it had basically got unmaintainable! At one point it also supported mutter, but I have stripped all that code out, as it was just too unwieldy. My intention is to get it to a point where it is reasonably consistent how the various platforms are supported, at least in terms of structure.

It's available packaged on the Raspberry Pi apt repo; I'll hopefully be able to update with this version in a week or so.
